Cilsil/Utils/ProgramState.cs

```diff
@@ -83,17 +93,39 @@ public class ProgramState
         /// </summary>
         public Dictionary<int, BoxedValueType> VariableIndexToBoxedValueType { get; }
 
+        /// <summary>
+        /// Previous expression registered by return node.
+        /// </summary>
+        private Expression PreviousReturnedExpression;
+
+        /// <summary>
+        /// Previous expression type registered by return node.
+        /// </summary>
+        private Typ PreviousReturnedType;
+
         /// <summary>
         /// True if there are remaining instructions to translate, and false otherwise.
         /// </summary>
         public bool HasInstruction => InstructionsStack.Count > 0;
 
+        /// <summary>
+        /// True if the peek instruction in instruction stack is at the beginning of an exception handling block, and false otherwise.
+        /// </summary>
+        public bool NextInstructionInExceptionHandlingBlock
+            => ExceptionBlockStartToEndOffsets.ContainsKey(InstructionsStack.Peek().Instruction.Offset);
+
         /// <summary>
         /// Program stack for the current method. Each entry is an expression and a type for the 
         /// expression.
         /// </summary>
         private ProgramStack ProgramStack;
 
+        /// <summary>
+        /// Dangling condition program stack for the current method. Each entry is an expression and a type for the 
+        /// expression.
+        /// </summary>
+        private ProgramStack DanglingConditionProgramStack;
```

Comment:
I have significant concerns regarding this modification. The whole point of the program stack is that it follows the program stack in the CIL -- deviation from this means that we are potentially violating what should be an accurate representation of what's going on in the bytecode. 

I would like to understand what specifically this modification is intended to accomplish -- I suspect that we can likely accomplish this in a simpler way which doesn't fundamentally change the program stack.

Reply:
I separate the in-prune condition program stack is because I observed some significant performance side effects that causes a lot of timeouts.

Reply:
I have investigated one program stack. You may find the latest results and details on our onenote. It seems that one program stack works fine without causing performance issue or crash. We did see for some repos we lost some null dereferences except one repo, which we gained around 12 null dereferences.

I also checked the cfg translation on some sample code like this one:
![image](https://user-images.githubusercontent.com/63893533/123135322-83240400-d406-11eb-8c83-362154c40ec9.png)
As you can see there is no difference after we adopt one program stack

Reply:
I will create a PR to adopt one program stack

Reply:
PR is [here](https://github.com/xi-liu-ds/infersharp/pull/8)